### ISSUE TITLE
Add spaces before UnitsML in MathML conversion(using Ox)

### ISSUE THREAD

**Issue by suleman-uzair** [plurimath/plurimath#222]
Add spaces before **Unitsml** in **MathML** output following the instructions mentioned in the https://github.com/metanorma/bipm-si-brochure/issues/222

Process **Ox** nodes in post conversion methods.

**Comment by opoudjis** [comment]
No, it hasn't worked. Interestingly, you've addressed the underlying issue correctly: there is now a correct space between numbers and units. But there is not a space being inserted between parenthesis and unit:

````
<formula id="_a146151f-3861-f5f0-640c-f7cb86f40f74_si-brochure-en" unnumbered="true"><stem type="MathML" block="true"><math xmlns="http://www.w3.org/1998/Math/MathML">
  <mstyle displaystyle="true">
    <mn>1</mn>
    <mo rspace="thickmathspace">⁢</mo>
    <mrow>
      <mstyle mathvariant="normal">
        <mi>A</mi>
      </mstyle>
    </mrow>
    <mo>=</mo>
    <mrow>
      <mo>(</mo>
      <mfrac>
        <mi>e</mi>
        <mrow>
          <mn>1.602 176 634</mn>
          <mo>×</mo>
          <msup>
            <mn>10</mn>
            <mrow>
              <mo>−</mo>
              <mn>19</mn>
            </mrow>
          </msup>
        </mrow>
      </mfrac>
      <mo>)</mo>
    </mrow>
    <mrow>
      <msup>
        <mstyle mathvariant="normal">
          <mi>s</mi>
        </mstyle>
        <mrow>
          <mo>−</mo>
          <mn>1</mn>
        </mrow>
      </msup>
    </mrow>
  </mstyle>
</math>
<asciimath>1 "unitsml(A)" = (e/(1.602176634 xx 10^(-19))) "unitsml(s^(-1))"</asciimath></stem></formula>
````

You can see that when `1 "unitsml(A)"` is processed, the result is the correct 

````
    <mn>1</mn>
    <mo rspace="thickmathspace">⁢</mo>
    <mrow>
      <mstyle mathvariant="normal">
        <mi>A</mi>
      </mstyle>
    </mrow>
````

But we also expect a space before `"unitsml(s^(-1))"`: what precedes it, within parentheses, is also considered a numerical expression.

**Comment by opoudjis** [comment]
So my rule expressed in the source ticket is wrong:

> * If the unitsml is preceded by mi or mn, and the unitsml contains a number or letter, they are separated by `<mo rspace='thickmathspace'>&#x2062;</mo>`. So `<mn>9</mn><mi mathvariant='normal'>m</mi>` (unitsml contains a letter, m, and is preceded by a number, mn) renders as `<mn>9</mn><mo rspace='thickmathspace'>&#x2062;</mo><mi mathvariant='normal'>m</mi>`, _9 m_.
> *  If the unitsml is preceded by mi or mn, and the unitsml contains a number or letter (e.g. °), it gets `<mo>&#x2062;</mo>`. I don't know now how that works, but without the rspace, the separation collapses in MathML to not showing a space: `<mn>9</mn><mo>&#x2062;</mo><mi>°</mi>` renders apparently as just _9°_.
> * If the unitsml is preceded by anything else (or nothing), it gets "".


The third rule is not right: let me think on what the right rule is...

**Comment by opoudjis** [comment]
The right rule is, if the preceding node *is or contains* `mi` or `mn`.

**Comment by opoudjis** [comment]
> if the preceding node is or contains mi or mn.

Should have specified: "contains *recursively*"

**Comment by suleman-uzair** [comment]
Fixed in [PR #240](https://github.com/plurimath/plurimath/pull/240)

**Comment by suleman-uzair** [comment]
@opoudjis Just released version `0.8.2`.

**Comment by opoudjis** [comment]
Thank you, that means this will be in place for Monday's release.
